tests/utils/tensorflow/test_model_data_utils.py

```diff
@@ -0,0 +1,138 @@
+import pytest
+import scipy.sparse
+import numpy as np
+import copy
+
+from rasa.utils.tensorflow import model_data_utils
+from rasa.utils.features import Features
+from rasa.nlu.constants import INTENT, ACTION_NAME
+from rasa.utils.tensorflow.constants import SENTENCE
+
+
+def test_create_zero_features():
+    # DENSE FEATURES
+    dense_feature_sentence = np.zeros((1, 100))
+    dense_feature_sentence[0, 50] = 1
+    dense_feature_sentence_features = Features(
+        features=dense_feature_sentence,
+        attribute=INTENT,
+        feature_type=SENTENCE,
+        origin=[],
+    )
+    features = [[None, None, [dense_feature_sentence_features]]]
+
+    zero_features = model_data_utils.create_zero_features(features)
+    assert len(zero_features) == 1
+    assert zero_features[0].is_dense()
+    assert (zero_features[0].features == np.zeros((1, 100))).all()
+
+    # SPARSE FEATURES
+    sparse_feature_sentence = scipy.sparse.coo_matrix(dense_feature_sentence)
+    sparse_feature_sentence_features = Features(
+        features=sparse_feature_sentence,
+        attribute=INTENT,
+        feature_type=SENTENCE,
+        origin=[],
+    )
+    features = [[None, None, [sparse_feature_sentence_features]]]
+    zero_features = model_data_utils.create_zero_features(features)
+    assert len(zero_features) == 1
+    assert zero_features[0].is_sparse()
+    assert (zero_features[0].features != scipy.sparse.coo_matrix((1, 100))).nnz == 0
+
+
+def test_surface_attributes():
+    intent_features = np.zeros((1, 100))
+    intent_features[0, 50] = 1

```

Comment:
You are not using the intent features later on in the test. Ideally you check if the features are still the same.